src/Raven.Server/NotificationCenter/NotificationsStorage.cs

```diff
@@ -226,23 +226,27 @@ private NotificationTableValue Get(string id, JsonOperationContext context, Rave
 
         public bool Delete(string id, RavenTransaction existingTransaction = null)
         {
-            if (Logger.IsInfoEnabled)
-                Logger.Info($"Deleting notification '{id}'.");
+            bool deleteResult;
 
             if (existingTransaction != null)
             {
-                return DeleteFromTable(existingTransaction);
+                deleteResult = DeleteFromTable(existingTransaction);
             }
-
-            bool deleteResult;
-
-            using (_contextPool.AllocateOperationContext(out TransactionOperationContext context))
-            using (var tx = context.OpenWriteTransaction())
+            else
             {
-                deleteResult = DeleteFromTable(tx);
-                tx.Commit();
+                using (_contextPool.AllocateOperationContext(out TransactionOperationContext context))
+                using (var tx = context.OpenWriteTransaction())
+                {
+                    deleteResult = DeleteFromTable(tx);
+                    tx.Commit();
+                }
             }
 
+            if (deleteResult && Logger.IsInfoEnabled)
+                Logger.Info($"Deleted notification '{id}'.");
```

Comment:
Shouldn't we log a bit more here? What _was_ the notification, who deleted it, etc?
Why are we recording this, anyway?